test/test_connectorcomp.jl

```diff
@@ -35,109 +21,146 @@ end
     end
 end
 
-m = Model()
-set_dimension!(m, :time, 2000:3000)
-nsteps = Mimi.dim_count(m.md, :time)
+years = 2000:2010
+late_start = 2005
+dim = Mimi.Dimension(years)
 
-add_comp!(m, ShortComponent; first=2100)
-add_comp!(m, Mimi.ConnectorCompVector, :MyConnector) # Rename component in this model
-add_comp!(m, LongComponent; first=2000)
+#------------------------------------------------------------------------------
+#  1. Manual way of using ConnectorComp
+#       TODO: are we no longer allowing users to add ConnectorComps manually?
+#       Currently deprecated because of checks for full/backup data during
+#       calls to connect_param!
+#------------------------------------------------------------------------------
```

Comment:
@davidanthoff 
For now I've commented out this test for the case of manually adding a ConnectorComp to a model. I added the error to `connect_param!` that will throw an error if a connection is made from a longer component to a shorter component without providing backup data. This therefore doesn't allow a user to add a ConnectorComp explicitly, because then if they call `connect_param(m, :ConnectorComp, :input, :Short, :var)` the new error will be thrown. I can add a work around for this so that the error is not thrown if the component is a ConnectorComp, or we can just choose to not support the manual use of ConnectorComps. Which do you prefer?